var/spack/repos/builtin/packages/py-cwl-utils/package.py

```diff
@@ -16,14 +16,27 @@ class PyCwlUtils(PythonPackage):
 
     license("Apache-2.0")
 
+    version("0.37", sha256="7b69c948f8593fdf44b44852bd8ef94c666736ce0ac12cf6e66e2a72ad16a773")
     version("0.21", sha256="583f05010f7572f3a69310325472ccb6efc2db7f43dc6428d03552e0ffcbaaf9")
 
     depends_on("python@3.6:", type=("build", "run"))
+    depends_on("python@3.8:", when="@0.29:", type=("build", "run"))
     depends_on("py-setuptools", type="build")
 
     depends_on("py-cwl-upgrader@1.2.3:", type=("build", "run"))
     depends_on("py-packaging", type=("build", "run"))
     depends_on("py-rdflib", type=("build", "run"))
     depends_on("py-requests", type=("build", "run"))
     depends_on("py-cachecontrol", type=("build", "run"))
-    depends_on("py-schema-salad@8.3.20220825114525:8", type=("build", "run"))
+    depends_on("py-schema-salad@8.3.20220825114525:8", when="@:0.31", type=("build", "run"))
+    depends_on("py-schema-salad@8.8.20250205075315:8", when="@0.37", type=("build", "run"))
```

Comment:
Should the gap from 0.32:0.36 be filled?

Would it be useful to have `@0.37:` as open ended range for the second line?

Reply:
I can fill the gap from 0.32:0.36, if necessary. I'll just need to add the versions.

I ended at ```@0.37``` because that is the latest version available on PyPI, but I do agree it's better to make the range open ended.

Reply:
Well, let me clarify that request. It's not really necessary to add the intermediate versions; I meant only the dependencies. Someone may decide to install `py-cwl-utils@=0.36` and spack would install it (spack would warn about a missing checksum). But, it would not depend on `py-schema-salad` at all after this change, so it would fail to install. A better approach is to make sure that the necessary dependencies are always satisfied, even for the versions not listed. In this case, for example, you can likely have the dependency for 0.37 just start at 0.32:
```suggestion
    depends_on("py-schema-salad@8.3.20220825114525:8", when="@:0.31", type=("build", "run"))
    # intermediate versions 0.32:0.36 may not require 8.8, but should work with this stricter requirement
    depends_on("py-schema-salad@8.8.20250205075315:8", when="@0.32:", type=("build", "run"))
```